src/common/services/UserService/UserService.ts

```diff
@@ -24,6 +24,35 @@ const createUser = async (values: CreateUserType): Promise<Response> => {
   });
 };
 
+const readUsers = async (params: ReadUsersQueryType): Promise<ReadUsersResponseType> => {
+  return await fetch(
+    `${import.meta.env.VITE_API_URL}/admin/users?size=${params?.query?.size === undefined ? '' : params?.query?.size}&page=${
+      params?.query?.page === undefined ? '' : params?.query?.page
+    }&sort=${params?.query?.sort === undefined ? '' : params?.query?.sort[0]},${params?.query?.sort === undefined ? '' : params?.query?.sort[1]}`
+  )
+    .then((response) => {
+      if (response !== null && response.ok) {
+        return response.json();
+      } else {
+        throw new FetchError("Couldn't fetch users");
+      }
+    })
+    .then((data: ReadUsersResponseType) => {
+      return data;
+    });
+};
+
```

Comment:
- [ ] Please use `async`/`await` syntax
- [ ] You are nesting so many ternary operators, what makes this code hard to read. I think you can create two separated `fetchQuerry` if the are any parameters or if there is no params
```typescript
let fetchQuery = `${import.meta.env.VITE_API_URL}/admin/users`
if (params) {
  addQueryParams(fetchQuery, params);
}
```
>`addQueryParams` can be common method for all kind of requestes